src/System.Devices.Gpio/System.Devices.Gpio/I2c/UnixI2cDevice.cs

```diff
@@ -291,7 +291,7 @@ private unsafe void FileInterfaceTransfer(byte[] buffer, TrasnferKind kind)
             int ret = ioctl(_deviceFileDescriptor, (uint)I2cSettings.I2C_SLAVE_FORCE, (ulong)_settings.DeviceAddress);
             if (ret < 0)
             {
-                throw new GpioException("Error performing I2c data transfer");
+                throw new IOException("Error performing I2c data transfer");
```

Comment:
even if you're not going to pretty-print the errno it would be useful to include ```ret``` in these exception messages for better diagnostics.